### PR TITLE
Add JsonTodo for converting Todos into Json objects

### LINKED ISSUE #4: Todo as JsonObject
The output should be a JsonArray with Todos as JsonObject. For Json, we should use JSON-P:

```java
<dependency>
    <groupId>org.glassfish</groupId>
    <artifactId>javax.json</artifactId>
    <version>1.1.4</version>
</dependency>
```

We will need AbstractJsonObject and AbstractJsonArray in order to implement polymorphism. Have a look at these classes: https://github.com/self-xdsd/self-web/tree/master/src/main/java/com/selfxdsd/selfweb/api/output. It should be the same pattern.

### PR DESCRIPTION
**Pull request by lukacupic**
Fixes #4 

Left puzzle for continuing implementation of Todos.

### CODE REVIEW

**Comment by charlesmike** [comment]
@lukacupic thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by charlesmike** [comment]
@amihaiemil I couldn't find any assignee for this task. This is either because there are no contributors with role ``REV`` available or because the project does not have enough funds.

Please, make sure there is at least one available contributor with the required role and the project can afford to pay them.

**Comment by amihaiemil** [comment]
@rultor merge it please

File: src/test/java/com/selfxdsd/todocli/JsonTodoTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import javax.json.JsonObject;
+
+/**
+ * Unit tests for {@link JsonTodo}.
+ *
+ * @version $Id$
+ * @since 0.0.1
+ */
+public class JsonTodoTest {
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/todocli/JsonTodoTest.java:14]: @lukacupic class should be final, always

File: src/main/java/com/selfxdsd/todocli/JsonTodo.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.selfxdsd.todocli;
+
+import javax.json.Json;
+
+/**
+ * Todo as JsonObject.
+ *
+ * @author lukacupic
+ */
+public class JsonTodo extends AbstractJsonObject {
```
Comment (amihaiemil) [inline src/main/java/com/selfxdsd/todocli/JsonTodo.java:10]: @lukacupic make it final please :D

File: src/main/java/com/selfxdsd/todocli/JsonTodo.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    /**
+     * Ctor.
+     *
+     * @param todo Todo to be converted to JsonObject.
+     */
+    public JsonTodo(final Todo todo) {
+        super(Json.createObjectBuilder()
+                // @todo #4:5min Uncomment the line below once #10 is merged
+//                .add("id", todo.getID())
```
Comment (amihaiemil) [inline src/main/java/com/selfxdsd/todocli/JsonTodo.java:20]: @lukacupic I think checkstyle complains about this comment. Just remove this line and write the todo in the JavaDoc of the method. "Add id attribute when #10 is finished"

File: src/main/java/com/selfxdsd/todocli/Todo.java [hunk 4]
```diff
@@ -22,12 +22,16 @@
  */
 package com.selfxdsd.todocli;
 
+// @todo #4:60min  Add interface Todos which should represent an iterable of Todo objects.
+//  Afterwards, add class JsonTodos, which should represent the TODOs as JsonArray.
```
Comment (amihaiemil) [inline src/main/java/com/selfxdsd/todocli/Todo.java]: @lukacupic It's better if you add todos in JavaDocs. Move this Todo to the class' JavaDoc. I'm pretty sure Checkstyle complains about it.

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@lukacupic Checkstyle has some complaits here too :D

**Pull request merged**